gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/privacySettings/js/AddAuthorizationDialog.js

```diff
@@ -0,0 +1,81 @@
+/*!
+GPII OAuth2 Add authorization dialog
+
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+The research leading to these results has received funding from the European Union's
+Seventh Framework Programme (FP7/2007-2013) under grant agreement no. 289016.
+
+You may obtain a copy of the License at
+https://github.com/GPII/universal/blob/master/LICENSE.txt
+*/
+
+// Declare dependencies
+/* global fluid, gpii, jQuery */
+
+(function ($, fluid) {
+    "use strict";
+
+    fluid.defaults("gpii.oauth2.addAuthorizationDialog", {
+        gradeNames: ["gpii.oauth2.privacySettingsDialog"],
+        requestInfos: {
+            addAuthorization: {
+                url: "/authorizations",
+                type: "post"
+            }
+        },
+        events: {
+            authorizationAdded: null
+        },
+        listeners: {
+            "onCreate.setInitialSelectedPrefs": {
+                listener: "{that}.setInitialSelectedPrefs",
+                args: [ {} ]
+            },
+            "onDone.sendAddAuthorization": {
+                listener: "{that}.sendAddAuthorization"
+            }
+        },
+        invokers: {
+            selectionsAsServerModel: {
+                funcName: "gpii.oauth2.addAuthorizationDialog.selectionsAsServerModel",
+                args: ["{that}.selectionTree"]
+            },
+            fireAuthorizationAdded: {
+                "this": "{that}.events.authorizationAdded",
+                method: "fire"
+            },
+            sendAddAuthorization: {
+                funcName: "gpii.oauth2.ajax",
```

Comment:
This is far too much material to write declaratively, with two nested stages of expansion! Please factor out into a standard function named, for example, gpii.oauth2.addAuthorizationDialog.sendAddAuthorization

We should add a TODO to update to the dataSource system once it is migrated out of Kettle up into Infusion